package.json

```diff
@@ -42,7 +42,7 @@
     "sharp": "0.33.3",
     "tailwind-merge": "2.5.2",
     "tailwindcss": "3.4.10",
-    "typescript": "5.5.4",
+    "typescript": "5.4.5",
```

Comment:
Tiputin Typescriptin version 5.5.4 -> 5.4.5. Tuo oli viimeisin kunnon päivitys, joka ei ollut >5.5.0. [npmjs.com:n lista](https://www.npmjs.com/package/typescript?activeTab=versions)

Ajaessa `npm run lint:fix` tai `npm run lint` tuli tämmöinen virheilmoitus:

=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=4.7.4 <5.5.0

YOUR TYPESCRIPT VERSION: 5.5.4

Please only submit bug reports when using the officially supported version.

=============

Ei ollut apua vaikka `@typescript-eslint/typescript-estree`-paketin asensikin, sama virheilmoitus.


Sitten seuraava ongelma koskee ESLint 9:iä. Eli kun ESLintin päivittää (8.57.0 -> 9.9.1) tulee seuraava ongelma:

`Could not resolve dependency: peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8" from eslint-plugin-import@2.29.1 
node_modules/eslint-plugin-import dev eslint-plugin-import@"2.29.1" from the root project`

Tätäkään en saanut korjattua

Myöskin ESLint 9:n ja Next Lintin kanssa on ollut hiukan ongelmaa [https://github.com/vercel/next.js/issues/64409](https://github.com/vercel/next.js/issues/64409)

`eslint-config-next`-paketille oli tullut päivitys, mutta kyseisestä päivityksestä en löytänyt tietoa


Onko siis ihan ok mennä tällä tavalla nyt eteenpäin?

Reply:
Koska Next.js ei vielä tue eslint v9, niin tuo mitä teit on paras ratkaisu. 

> Sitten seuraava ongelma koskee ESLint 9:iä. Eli kun ESLintin päivittää (8.57.0 -> 9.9.1) tulee seuraava ongelma:
> 
> `Could not resolve dependency: peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8" from eslint-plugin-import@2.29.1 node_modules/eslint-plugin-import dev eslint-plugin-import@"2.29.1" from the root project`
> 
> Tätäkään en saanut korjattua

Sitä ei oikein saakaan, kun ainakaan `eslint-plugin-import` [ei tue vielä lainkaan eslint v9](https://github.com/import-js/eslint-plugin-import/issues/2948). Eli käytetään eslint v8 ja tuo TypeScriptin versio kunnes v9 tuki laajenee.